pkg/component/shootsystem/shootsystem.go

```diff
@@ -478,3 +480,15 @@ func (s *shootSystem) readOnlyRBACResources() []client.Object {
 
 	return []client.Object{clusterRole, clusterRoleBinding}
 }
+
+func (s *shootSystem) isEncryptedResource(resource, group string) bool {
+	if group == corev1.SchemeGroupVersion.Group {
+		// cores resource can also be present as '<resource>.' For example, 'configmaps' as 'configmaps.'
+		coreResourceName := fmt.Sprintf("%s.", resource)
```

Comment:
Also in discovery result?

Reply:
I am adding `.` to the name of the resource in the discovery result and checking if it's present in the list.

Reply:
IMHO, this `.` notation is a bit risky because we need an obscure handling for it in different places. WDYT if we either always trim the `.` from the API object or either normalize the list centrally in the code?

Reply:
This is the specification provided by the user. `configmaps.` and `configmaps` both are allowed by the kube-apiserver.
If we normalize it in the reconciliation flow, then while setting the `status.encryptedResources`, this might be an issue. Do you want to add validation to prevent specifying the former?

Reply:
We can pass a normalized encrypted resources list to this component, but I wouldn't prefer altering the encryption config for the kube-apiserver.

Reply:
Can you explain why? What's the downside?

Reply:
Since we compare the resources many times in the flow. If we normalize the spec then we would have to normalize the status also and keep them in `b.Shoot.ResourcesToEncrypt` and `b.Shoot.EncryptedResources`.
And while setting the status, we use the original spec itself.

Reply:
I pushed https://github.com/gardener/gardener/pull/8964/commits/a58d42b2426b93dbb651f623c889a5e391acfe9f, PTAL!